authors/apps/articles/models.py

```diff
@@ -43,3 +47,73 @@ class ArticleImage(models.Model):
 
     class Meta:
         ordering = ('-date_created',)
+
+class LikeArticles(models.Model):
+    user = models.ForeignKey(
+        User,
+        related_name='liked_by',
+        on_delete=models.CASCADE
+    )
+    article = models.ForeignKey(
+        Article,
+        to_field='slug',
+        related_name='liked',
+        db_column='article',
+        on_delete=models.CASCADE
+    )
+    likes = models.IntegerField(null=True)
+    dislikes = models.IntegerField(null=True)
+    created_on = models.DateTimeField(auto_now=True)
+    class Meta:
+        ordering = ('created_on',)
+
+    @staticmethod
+    def retrieve_article(slug):
+        """Method gets a single article"""
+        article = get_object_or_404(Article, slug=slug)
+        return article
+
+    @staticmethod
+    def like_by_user(user, slug, ArticleSerializer, value):
+        """This method holds logic for like, dislike, or revoking"""
+        try:
+            likes = LikeArticles.objects.filter(user=user, article=slug)
+            article = LikeArticles.retrieve_article(slug=slug)
+        except:
+            APIException.status_code = status.HTTP_404_NOT_FOUND
+            raise APIException(
+                {
+                    'article': {
+                        'message': 'Article requested does not exist'
+                    }
+                }
+            )
+        if not likes:
```

Comment:
Just a minor suggestion that wouldn't hinder merging. I think this would more readable if we checked for if likes and returned the error message then have the rest of the logic